metashare/api/models.py

```diff
@@ -289,6 +294,23 @@ def save(self, *args, **kwargs):
 
         return ret
 
+    def get_login_url(self):
+        org_config = refresh_access_token(
+            config=self.config, org_name="dev", login_url=self.login_url
+        )
+        self.config = org_config.config
+        self.save()
+        session = jwt_session(
+            settings.SF_CLIENT_ID,
+            settings.SF_CLIENT_KEY,
+            org_config.username,
+            url=self.login_url,
+        )
+        return urljoin(
+            str(session["instance_url"]),
+            "secur/frontdoor.jsp?sid={}".format(session["access_token"]),
+        )
```

Comment:
So here's how I would implement this method:

    def get_login_url(self):
        org_config = refresh_access_token(config=self.config, org_name="dev", login_url=self.login_url)
        return org_config.start_url

Reply:
And actually, I just merged the fixed that MetaShare needs in CumulusCI's jwt_session: https://github.com/SFDO-Tooling/CumulusCI/pull/1256

With that change, if you use cumulusci from master you should be able to do:

    org_config = OrgConfig(self.config, "dev")
    org_config.refresh_oauth_token(None)
    return org_config.start_url

In other words I think you'll be able to use the standard `refresh_oauth_token` method, and you won't need to store the login_url any more because CumulusCI's jwt_session will derive it from the instance_url stored in the org config.

Reply:
It would probably make sense to give ScratchOrg a method like this:

    def get_refreshed_org_config(self):
        org_config = OrgConfig(self.config, "dev")
        org_config.refresh_oauth_token(None)
        return org_config

and make use of it wherever you need an org config with a fresh access token

Reply:
@davisagli that `OrgConfig` method will depend on a release, right?

Reply:
@wlonk yes, or using cumulusci master, like I said

Reply:
Right.

Reply:
So I had to take some of the guts out of your `OrgConfig.refresh_oauth_token` method and use that directly, because of the limitations on how we can store and access the hub key. But take a look, see what you think of it. I am not keen on it using private methods directly, but it works.

Reply:
@wlonk this looks reasonable to me now, but I think we could make MetaShare's settings file set os.environ['SFDX_HUB_KEY'] based on decoding the newlines in os.environ["SF_CLIENT_KEY"]